openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/ContainerRepository.java

```diff
@@ -0,0 +1,207 @@
+      populateDataModelColumnTags(fields.contains(FIELD_TAGS), container.getDataModel().getColumns());
+    }
+    container.setFollowers(fields.contains(FIELD_FOLLOWERS) ? getFollowers(container) : null);
+    return container;
+  }
+
+  private void populateDataModelColumnTags(boolean setTags, List<Column> columns) {
+    for (Column c : listOrEmpty(columns)) {
+      c.setTags(setTags ? getTags(c.getFullyQualifiedName()) : null);
+      populateDataModelColumnTags(setTags, c.getChildren());
+    }
+  }
+
+  private EntityReference getParentContainer(Container container) throws IOException {
+    if (container == null) return null;
+    return getFromEntityRef(container.getId(), Relationship.CONTAINS, CONTAINER, false);
+  }
+
+  private void setDefaultFields(Container container) throws IOException {
+    EntityReference parentObjectStoreServiceRef =
+        getFromEntityRef(container.getId(), Relationship.CONTAINS, OBJECT_STORE_SERVICE, true);
+    container.withObjectStoreService(parentObjectStoreServiceRef);
+  }
+
+  private List<EntityReference> getChildrenContainers(Container container) throws IOException {
+    if (container == null) {
+      return Collections.emptyList();
+    }
+    List<CollectionDAO.EntityRelationshipRecord> childContainerIds =
+        findTo(container.getId(), CONTAINER, Relationship.CONTAINS, CONTAINER);
+    return EntityUtil.populateEntityReferences(childContainerIds, CONTAINER);
+  }
+
+  @Override
+  public void setFullyQualifiedName(Container container) throws IOException {
+    if (container.getParent() != null) {
+      container.setFullyQualifiedName(
+          FullyQualifiedName.add(container.getParent().getFullyQualifiedName(), container.getName()));
+    } else {
+      container.setFullyQualifiedName(
+          FullyQualifiedName.add(container.getObjectStoreService().getName(), container.getName()));
+    }
+    if (container.getDataModel() != null) {
+      setColumnFQN(container.getFullyQualifiedName(), container.getDataModel().getColumns());
+    }
+  }
+
+  private void setColumnFQN(String parentFQN, List<Column> columns) {
+    columns.forEach(
+        c -> {
+          String columnFqn = FullyQualifiedName.add(parentFQN, c.getName());
+          c.setFullyQualifiedName(columnFqn);
+          if (c.getChildren() != null) {
+            setColumnFQN(columnFqn, c.getChildren());
+          }
+        });
+  }
+
+  @Override
+  public void prepare(Container container) throws IOException {
+    // the objectStoreService is not fully filled in terms of props - go to the db and get it in full and re-set it
+    ObjectStoreService objectStoreService =
+        Entity.getEntity(container.getObjectStoreService(), "", Include.NON_DELETED);
+    container.withObjectStoreService(objectStoreService.getEntityReference());
+
+    // validate parent reference
+    if (container.getParent() != null) {
+      daoCollection.containerDAO().findEntityReferenceById(container.getParent().getId());
+    }
```

Comment:
We call `setFields` before `prepare`, so we already get this information from the `entity_relationships` table. In case of any error, it would imply a DB inconsistency. IMHO, this extra query is redundant. I would expect some validation here.

Reply:
Makes sense, will remove it.